spec/lyber_core/robots/robot_spec.rb

```diff
@@ -1,103 +1,137 @@
-      logged = capture_stdout do
-        TestRobotWithNote.perform druid
+    context 'when correct state and a note returned' do
+      let(:test_class) { test_robot_with_note }
+      it "updates workflow to 'completed' and sets a custom note" do
+        expect(Dor::WorkflowService).to receive(:update_workflow_status).with('dor', druid, wf_name, step_name, 'completed',
+                                                                              elapsed: Float,
+                                                                              note: 'some note to pass back to workflow')
+        expect(logged).to match(/#{druid} processing/).and match(/work done\!/)
       end
-      expect(logged).to match /#{druid} processing/
-      expect(logged).to match /work done\!/
     end
 
-    it "updates workflow to 'completed' and sets a custom note if work processes and returns the object with the correct state and a note" do
-      expect(Dor::WorkflowService).to receive(:get_workflow_status).with('dor', druid, wf_name, step_name).and_return('queued')
-      expect(Dor::WorkflowService).to receive(:update_workflow_status).with('dor', druid, wf_name, step_name, 'skipped',
-                                                                            elapsed: an_instance_of(Float),
-                                                                            note: 'some note to pass back to workflow')
-      logged = capture_stdout do
-        TestRobotWithNoteAndSkip.perform druid
+    context 'when skipped state and a note returned' do
+      let(:test_class) { test_robot_with_note_and_skip }
+      it "updates workflow to 'skipped' and sets a custom note" do
+        expect(Dor::WorkflowService).to receive(:update_workflow_status).with('dor', druid, wf_name, step_name, 'skipped',
+                                                                              elapsed: Float,
+                                                                              note: 'some note to pass back to workflow')
+        expect(logged).to match(/#{druid} processing/).and match(/work done\!/)
       end
-      expect(logged).to match /#{druid} processing/
-      expect(logged).to match /work done\!/
     end
 
-    it "updates workflow to 'skipped' using a ReturnState constant" do
-      expect(Dor::WorkflowService).to receive(:get_workflow_status).with('dor', druid, wf_name, step_name).and_return('queued')
-      expect(Dor::WorkflowService).to receive(:update_workflow_status).with('dor', druid, wf_name, step_name, 'skipped',
-                                                                            elapsed: an_instance_of(Float),
-                                                                            note: Socket.gethostname)
-      logged = capture_stdout do
-        TestRobotWithConstantState.perform druid
+    describe 'using a ReturnState constant' do
```

Comment:
any reason this one is describe instead of context?

Reply:
don't care either way